infra/Pos/Slotting/Types.hs

```diff
@@ -23,14 +33,65 @@ data EpochSlottingData = EpochSlottingData
 instance NFData EpochSlottingData
 
 -- | Data necessary for slotting to work which is basically part of GState.
-data SlottingData = SlottingData
-    { sdPenult      :: !EpochSlottingData
-    -- ^ 'EpochSlottingData' for penult (i. e. 'sdPenultEpoch') epoch.
-    , sdLast        :: !EpochSlottingData
-    -- ^ 'EpochSlottingData' for last epoch.
-    -- Last means 'sdPenultEpoch + 1'.
-    , sdPenultEpoch :: !EpochIndex
-    -- ^ Penult epoch for which 'EpochSlottingData' is known.
-    } deriving (Eq, Show, Generic)
+-- TODO: Reflect the fact that this will always have 2+ elements
+-- TODO: Make this a newtype wrapper
```

Comment:
Yes, can you do both TODOs within this issue? I see two approaches:
1. Make it `newtype` over `HashMap EpochIndex EpochSlottingData` and add some unsafe functions to this module which will use `error` if the invariant is violated. So that external code can use functions like `getLastEpochIndex` or `getLastEpochSlottingData` and not worry about cases where it doesn't exist (this module should be responsible for it).
Note that I find it important to use `error` rather than default values like `0`, because such cases indicate invariants violation and shouldn't be hidden behind default values.
2. Make it `data` and store two last `EpochSlottingData`s explicitly. Probably it will require you to write more code, but it will make the code safer and we will have less partial functions.